### DISCUSSION TITLE
Wakepy and pyinstaller - does it works?

### DISCUSSION THREAD

**Discussion by folkien** [fohrloop/wakepy#275] [Q&A]
Hello,

I have a python program packed with pyinstaller which uses wakepy. After packing for windows 10 .exe program is halted when wakepy `with keep.running()` is called. Have you tested it with pyinstaller?

**Comment by fohrloop** [comment]
Hi @folkien , there is nothing so special in wakepy which would make it not possible to use it in a packaged executable. One thing to consider, though: The current (0.7.2) version uses still parts from a subpackage called `_deprecated`. As this has dynamical (runtime) imports, pyinstaller and alike do not see the import and cannot automatically add the files to the bundle. This means that those files need to be added to the bundle manually. 

See also https://github.com/fohrloop/wakepy/issues/49

Note that the `_deprecated` subpackage is, as the name also implies, deprecated and will be removed in a future version of wakepy. In the future, I hope the dynamical imports can be avoided altogether. 

Hope this helps!

Niko

**Reply by fohrloop** [reply]
Related: https://github.com/wakepy/wakepy/issues/318
